app/concepts/api/v1/users/registrations/contract/validate.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Api::V1::Users::Registrations::Contract
+  class Validate < ApplicationContract
+    params do
+      required(:email).filled(:string)
+      required(:password).filled(:string)
+    end
+
+    rule(:email) do
+      key.failure('has allready been taken') if User.exists?(email: value)
```

Comment:
What is pure text doing here? Please move same text snippets to i18n localizations.

Reply:
Replaced with I18n